remoting/etcd3/client.go

```diff
@@ -127,6 +127,8 @@ type Client struct {
 // NewClient create a client instance with name, endpoints etc.
 func NewClient(name string, endpoints []string, timeout time.Duration, heartbeat int) (*Client, error) {
 	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
hey, guy, do u know what are you doing? If you invoke cancel,  the cancel will be useless on line 149.